posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -223,6 +240,15 @@ async def insert_into_s3_activity(inputs: S3InsertInputs):
         )
 
 
+def create_temporary_file(compression: Literal["none", "gzip"]):
+    """Create a temporary file with a random name."""
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
This means the file won't be deleted upon close. Why not write a new context manager to handle it and ensure it's closed to avoid leaks?

Reply:
It also keeps the file management constrained here, and we don't have to worry about calling `os.remove` or `local_results_file.close`.

Reply:
> This means the file won't be deleted upon close. Why not write a new context manager to handle it and ensure it's closed to avoid leaks?

Yeah I did a context manager initially. But I couldn't quite get my head around how to handle to rotation of the file when we reach a certain limit 😅 

I'll give it another go as would be nice to have a solid way to handle this, both in s3 and snowflake.

Reply:
Ok I think I've made it a little nicer now(?)

What do you think?

Reply:
(there is probably some further abstraction that could be nice but I'm thinking hold off until it's absolutely necessary, maybe when unifying the snowflake implementation 🤔 )